python/telemetry/snet.py

```diff
@@ -168,10 +100,91 @@ def _decode(self, obj, context, path=None):
     'TargetData_ControlIsActive' / Int8ul  # flag, really
     )
 
+LTUFrameHeader = BitStruct(
+    'SrcId' / BitsInteger(7),
+    'DstId' / BitsInteger(7),
+    'FrCntTx' / BitsInteger(4),
+    'FrCntRx' / BitsInteger(4),
+    'SNR' / BitsInteger(4),
+    'AiTypeSrc' / BitsInteger(4),
+    'AiTypeDst' / BitsInteger(4),
+    'DfcId' / BitsInteger(2),
+    'Caller' / Flag,
+    'Arq' / Flag,
+    'PduTypeId' / Flag,
+    'BchRq' / Flag,
+    'Hailing' / Flag,
+    'UdFl1' / Flag,
+    'PduLength' / BitsInteger(10),
+    'CRC13' / BitsInteger(13),
+    'CRC5' / BitsInteger(5),
+    Padding(2)
+    )
+
+# For reference:
+#
+# To send mailbox messages
+# FCIDMajor = 56  (COM0)
+# FCIDSub = 34 || 134  (COM1)
+
+MailboxTelemetry = Struct(
+    'SysTime' / BitsInteger(4),
+    'MemDataType' / BitsInteger(1),
+    'MemData32' / Hex(GreedyBytes)
+    # 'MemData32' / LinearAdapter(32, Int8ub)
+    )
+
 
-snet = Struct(
-    'header' / SNETFrameHeader,
-    'telemetry' / Switch(lambda c: (c.header.FCIDMajor, c.header.FCIDSub), {
+class TimeAdapter(Adapter):
+    def _encode(self, obj, context, path=None):
+        d = int((obj - datetime.datetime(2000, 1, 1))*2)
+        return Container(
+            days=d.days,
+            milliseconds=d.seconds * 1000 + d.microseconds / 1000)
+
+    def _decode(self, obj, context, path=None):
+        return (datetime.datetime(2000, 1, 1)
+                + datetime.timedelta(seconds=float(obj)/2.0))
+
+
+TimeStamp = TimeAdapter(BitsInteger(32, swapped=True))
+
+PDUFrameHeaderExtension = BitStruct(
```

Comment:
Is there a reason for renaming `SNETFrameHeaderExtension` to `PDUFrameHeaderExtension` and `SNETFrameHeader` to `PDUFrameHeader`?

Reply:
I matched the names with what I saw (being named) in the implementation, so the deframer follows the name convention of how is defined internally and on university docs.

It's a nitpick from my side, probably yes, but I think when you get access to the SNET information and later check what is implemented in gr-satellites, you get an easier 1:1 relation to the different pieces and can connect the dots quickly.

I can roll back this change if it doesn't make sense, no problem for me.

Reply:
If the documentation is not publicly available, then I think the name change is not justified.

Reply:
I liked the new docs at the top of the file where the link was broken, could you please check and see if the rename makes sense now?

still, I don't have a problem rolling back these specific changes.

Reply:
Okay for the `SNETFrameHeaderExtension` -> `PDUFrameHeaderExtension` and `SNETFrameExtension` -> `PDUFrameHeader` rename, but that's the only thing that should show up in the diff.